easybuild/easyblocks/b/blis.py

```diff
@@ -83,7 +83,16 @@ def configure_step(self):
         if self.cfg['cpu_architecture'] == 'auto':
             failed_detect_str = r'Unable to automatically detect hardware type'
             if re.search(failed_detect_str, output):
-                raise EasyBuildError(failed_detect_str)
+                fallback_archs = {
+                    'zen5': 'zen3',
```

Comment:
Would it make sense to use the same fallback for `zen4`? BLIS 2.0 does not yet contain the kernels for `zen4` and `zen5`.
```suggestion
                    'zen4': 'zen3',
                    'zen5': 'zen3',
```

Reply:
For some reason, Zen 4 doesn’t fail currently (when testing on an AMD Ryzen 7800X3D). I can check what BLIS uses there…

Reply:
As mentioned, recent BLIS versions should automatically fall back to `zen3` for `zen4`. So we should be fine here. I don't think that we need to do changes to support deprecated toolchains.

```console
configure: hardware detection driver returned 'zen3'.
configure: checking configuration against contents of 'config_registry'.
configure: configuration 'zen3' is registered.
configure: 'zen3' is defined as having the following sub-configurations:
configure:    zen3
configure: which collectively require the following kernels:
configure:    zen3 zen2 zen haswell
configure: checking sub-configurations:
configure:   'zen3' is registered...and exists.
configure: checking sub-configurations' requisite kernels:
configure:   'zen3' kernels...exist.
configure:   'zen2' kernels...exist.
configure:   'zen' kernels...exist.
configure:   'haswell' kernels...exist.
configure: detected --prefix='/data/EasyBuild-develop/software/BLIS/2.0-GCC-14.3.0'.
```

Reply:
Ah, I didn't know that BLIS is detecting `zen4` as `zen3`. I agree, it makes perfect sense to only add the fallback for `zen5`.